pkg/client/client.go

```diff
@@ -0,0 +1,53 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ObjectKey identifies a Kubernetes Object.
+type ObjectKey = types.NamespacedName
+
+// Client defines CRUD operations for runtime object in the cluster
+type Client interface {
+	Create(ctx context.Context, obj runtime.Object) error
+	Get(ctx context.Context, key ObjectKey, objExisting runtime.Object) error
+	Update(ctx context.Context, obj runtime.Object) error
+}
+
+// crud is an implementation of the Client interface
+type crud struct {
```

Comment:
@aravindhp @tkashem @kevinrizza let me know if you guys can think of a better name for this struct.

Reply:
We don't expose a `delete` method at all, so `crud` doesn't make sense. But also I would prefer to use some kind of noun that describes what the struct does, rather than a [Nominalization](https://en.wikipedia.org/wiki/Nominalization).

I think there's nothing wrong with calling such a thing `client` or `kubeClient`, given that that's the name of the interface it is implementing.